metpy/plots/station_plot.py

```diff
@@ -279,7 +276,20 @@ def plot_barb(self, u, v, **kwargs):
         if self.barbs:
             self.barbs.remove()
 
-        self.barbs = self.ax.barbs(x, y, u, v, **defaults)
+        # Handle transforming our center points. CartoPy doesn't like 1D barbs
+        # TODO: This can be removed for cartopy > 0.14.3
+        if hasattr(self.ax, 'projection') and 'transform' in kwargs:
+            trans = kwargs['transform']
+            try:
+                kwargs['transform'] = trans._as_mpl_transform(self.ax)
+            except AttributeError:
```

Comment:
When could there be an AttributeError here given then conditions in the if?

Reply:
If the transform passed in is just a standard matplotlib transform rather than a cartopy projection. I've borrowed this directly from CartoPy's barbs code.